build/xtask/src/config.rs

```diff
@@ -719,3 +665,75 @@ impl BuildConfig<'_> {
         cmd
     }
 }
+
+////////////////////////////////////////////////////////////////////////////////
+
+fn read_and_flatten_toml(
+    cfg: &Path,
+    hasher: &mut DefaultHasher,
+    seen: &mut BTreeSet<PathBuf>,
+) -> Result<toml_edit::Document> {
+    use toml_patch::merge_toml_documents;
+
+    // Prevent diamond inheritance
+    if !seen.insert(cfg.to_owned()) {
+        bail!(
+            "{cfg:?} is inherited more than once; \
+             diamond dependencies are not allowed"
+        );
+    }
+    let cfg_contents = std::fs::read(&cfg)
+        .with_context(|| format!("could not read {}", cfg.display()))?;
+
+    // Accumulate the contents into the buildhash here, so that we hash both
+    // the inheritance file and the target (recursively, if necessary)
+    hasher.write(&cfg_contents);
+
+    let cfg_contents = std::str::from_utf8(&cfg_contents)
+        .context("failed to read manifest as UTF-8")?;
+
+    // Additive TOML file inheritance
+    let mut doc = cfg_contents
+        .parse::<toml_edit::Document>()
+        .context("failed to parse TOML file")?;
+    if let Some(inherited_from) = doc.remove("inherit") {
```

Comment:
Trivial nit - since this block ends with an "early" return (followed immediately by the normal return), I might use `if let` to bail out early in the "no inheritance" case, just to reduce the scope of the remainder:

```rust
let Some(inherited_from) = doc.remove("inherit") else {
    // No TOML inheritance involved; return the original document.
    return Ok(doc);
};
```

Reply:
Sure, done in 94bbd35474bff499551c8c24b58b05bbc6e1f395